src/planners/bidirectional.jl

```diff
@@ -0,0 +1,130 @@
+export BiDirectionalPlanner, BiGreedyPlanner, BiAStarPlanner
+
+"BiDirectionalPlanner best-first search planner."
+mutable struct BiDirectionalPlanner{F,B} <: Planner
+    forward::F
+    backward::B
+end
+
+BiDirectionalPlanner(;kwargs...) = BiDirectionalPlanner(
+    	ForwardPlanner(; kwargs...),
+    	BackwardPlanner(; kwargs...)
+    )
+
+"Bidirectional  Greedy best-first search, with cycle checking."
+BiGreedyPlanner(heuristic::Heuristic; kwargs...) =
+    BiDirectionalPlanner(
+    	ForwardPlanner(;heuristic=heuristic, g_mult=0, kwargs...),
+    	BackwardPlanner(;heuristic=heuristic, g_mult=0, kwargs...)
+    )
+
+
+"Bidirectional A* search."
+BiAStarPlanner(heuristic::Heuristic; kwargs...) =
+    BiDirectionalPlanner(
+    	ForwardPlanner(;heuristic=heuristic, kwargs...),
+    	BackwardPlanner(;heuristic=heuristic, kwargs...)
+    )
+
+function solve(planner::BiDirectionalPlanner,
+               domain::Domain, state::State, spec::Specification)
+
+    # Simplify goal specification
+    forward_spec = simplify_goal(spec, domain, state)
+    backward_spec = BackwardSearchGoal(spec, state)
+    # Precompute heuristic information
+    precompute!(planner.forward.heuristic, domain, state, forward_spec)
+    precompute!(planner.backward.heuristic, domain, state, backward_spec)
+    
+	forward_search_tree, forward_queue = init_forward(planner.forward, domain, state, spec)
+	backward_search_tree, backward_queue = init_backward(planner.backward, domain, state, spec)
+
+    status, node_id, count = search!(planner, domain, 
+    	forward_spec, forward_search_tree, forward_queue,
+    	backward_spec, backward_search_tree, backward_queue)
+
+    # Reconstruct plan and return solution
+    if status != :failure
+        forward_plan, forward_traj = reconstruct(node_id, forward_search_tree)
+        backward_plan, backward_traj = reconstruct(node_id, backward_search_tree)
+        reverse!(backward_plan); reverse!(backward_traj)
+        plan = vcat(forward_plan, backward_plan)
+        traj = vcat(forward_traj, backward_traj)
```

Comment:
So as mentioned in [this comment](https://github.com/JuliaPlanners/SymbolicPlanners.jl/pull/8#issuecomment-1345398608), the states explored by forward and backward search are actually different. This means you can't just concatenate the trajectories found by forward and backward search to get a single trajectory of complete states.

Instead, I think it's easier to just avoid storing the full trajectory as part of the `BiPathSearchSolution`. Alternatively, you could use the `StateTrajectorySimulator` to simulate the full trajectory, once you have the full plan.